gadget-container/entrypoint/entrypoint.go

```diff
@@ -0,0 +1,295 @@
+// Copyright 2019-2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"syscall"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/utils/host"
+	"golang.org/x/sys/unix"
+)
+
+var regex = regexp.MustCompile(`^1:name=systemd:.*/crio-[0-9a-f]*\.scope$`)
+
+func getPrettyName() (string, error) {
+	path := filepath.Join(host.HostRoot, "/etc/os-release")
+	file, err := os.Open(path)
+	if err != nil {
+		return "", fmt.Errorf("opening file %s: %w", path, err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		parts := strings.SplitN(line, "=", 2)
+		if len(parts) != 2 {
+			continue
+		}
+
+		if parts[0] == "PRETTY_NAME" {
+			return strings.Trim(parts[1], "\""), nil
+		}
+	}
+
+	err = scanner.Err()
+	if err != nil {
+		return "", fmt.Errorf("reading file %s: %w", path, err)
+	}
+
+	return "", fmt.Errorf("%s does not contain PRETTY_NAME", path)
+}
+
+func getKernelRelease() (string, error) {
+	uts := &unix.Utsname{}
+	if err := unix.Uname(uts); err != nil {
+		return "", fmt.Errorf("calling uname: %w", err)
+	}
+
+	return unix.ByteSliceToString(uts.Release[:]), nil
+}
+
+func copyFile(destination, source string, filemode fs.FileMode) error {
+	content, err := os.ReadFile(source)
+	if err != nil {
+		return fmt.Errorf("reading %s: %w", source, err)
+	}
+
+	err = os.WriteFile(destination, content, filemode)
+	if err != nil {
+		return fmt.Errorf("writing %s: %w", destination, err)
+	}
+
+	return nil
+}
+
+func installCRIOHooks() error {
+	log.Info("Installing hooks scripts on host...")
+
+	path := filepath.Join(host.HostRoot, "opt/hooks/oci")
+	err := os.MkdirAll(path, 0o755)
+	if err != nil {
+		return fmt.Errorf("creating %s: %w", path, err)
+	}
+
+	for _, file := range []string{"ocihookgadget", "prestart.sh", "poststop.sh"} {
+		log.Infof("Installing %s", file)
+
+		path := filepath.Join("/opt/hooks/oci", file)
+		destinationPath := filepath.Join(host.HostRoot, path)
+		err := copyFile(destinationPath, path, 0o640)
```

Comment:
Not related to this PR, but I noticed the path on the host is hardcoded to `/host/opt/hooks/oci` and I think it will be problematic when we start supporting several deployments of Inspektor Gadget in parallel with #2208.

cc @burak-ok